### PR TITLE
Catalog: update preview logic (VCF, text, Excel)

### PR DESCRIPTION
**Pull request by nl0**
## Summary

- [x] preview VCF files
- [x] use preview endpoint for text formats
- [x] preview CSV/TSV
- [x] use thumbnailer for images
- [x] support compression (`.gz`, `.bz2`)
- [x] get and parse `Content-Range` header

### CODE REVIEW

File: catalog/app/components/Preview/Preview.js [hunk 1]
```diff
@@ -15,10 +15,12 @@ const fallback = {
 };
 
 const loaderChain = [
+  loaders.Csv,
```
Comment (akarve) [inline catalog/app/components/Preview/Preview.js]: fine for now, but i question the casing. we have csv, CSV, Csv, etc. all in the same combined codebase

File: catalog/app/components/Preview/Preview.js [hunk 2]
```diff
@@ -15,10 +15,12 @@ const fallback = {
...
   loaders.Json,
   loaders.Markdown,
   loaders.Notebook,
   loaders.Parquet,
+  loaders.Vcf,
```
Comment (akarve) [inline catalog/app/components/Preview/Preview.js]: OK for separate PR, but we now support XLS and XLSX as well - works like .parquet (minus the info metadata)

File: catalog/app/components/Preview/Preview.js (hunk 1 above)
Comment (nl0) [reply catalog/app/components/Preview/Preview.js]: we use the js CamelCase (not sure if it's 100% precise term) convention here, abbreviations are considered words, so only the first letter should be capitalized, e.g. `JsonEncoder`. it's not 100% consistent across the codebase, but i'm trying to adhere to that convention.

File: catalog/app/components/Preview/Preview.js (hunk 2 above)
Comment (nl0) [reply catalog/app/components/Preview/Preview.js]: ok

**Pull request merged**